### PR TITLE
[FEATURE] first prototype of Annotation1DVerticalLineItem

### PR DESCRIPTION
**Pull request by JeeH-K**
# Description

A simple prototype of annotation item, Annotation1DVerticalLineItem

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

File: src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// --------------------------------------------------------------------------
+// $Maintainer: Jihyung Kim, Timo Sachsenberg $
+// $Authors: Jihyung Kim, Timo Sachsenberg $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/VISUAL/ANNOTATION/Annotation1DItem.h>
+#include <OpenMS/DATASTRUCTURES/DPosition.h>
+#include <vector>
```
Comment (cbielow) [inline src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h:39]: not needed

File: src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Jihyung Kim, Timo Sachsenberg $
+// $Authors: Jihyung Kim, Timo Sachsenberg $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/VISUAL/ANNOTATION/Annotation1DItem.h>
+#include <OpenMS/DATASTRUCTURES/DPosition.h>
```
Comment (cbielow) [inline src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h:38]: not needed

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+{
+
+  Annotation1DVerticalLineItem::Annotation1DVerticalLineItem(const double& x, const QColor& color, const QString & text) :
+      Annotation1DItem(text),
+      x_(x),
+      color_(color)
+  {
+  }
+
+  Annotation1DVerticalLineItem::Annotation1DVerticalLineItem(const Annotation1DVerticalLineItem & rhs) :
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp:53]: use ` = default;` instead?

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      Annotation1DItem(rhs)
+  {
+    x_ = rhs.x_;
+    color_ = rhs.color_;
+  }
+
+  Annotation1DVerticalLineItem::~Annotation1DVerticalLineItem()
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp:60]: also defaultable

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    QPoint start_p, end_p;
+    canvas->dataToWidget(x_, 0, start_p, flipped, true);
+    canvas->dataToWidget(x_, canvas->getDataRange().maxY(), end_p, flipped, true);
+
+    // draw line
+    painter.drawLine(start_p, end_p);
+
+    // compute bounding box on the specified painter
+    // TODO: implement proper bounding box calculation
+    if (canvas->isMzToXAxis())
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp:76]: code in if/else cases is identical.
This is probably a bug

File: src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+{
+  /** @brief An annotation item which represents a vertical line.
+          @see Annotation1DItem
+  */
+  class Annotation1DVerticalLineItem :
+      public Annotation1DItem
+  {
+  public:
+    /// Constructor
+    Annotation1DVerticalLineItem(const double& x, const QColor& color, const QString & tex="");
```
Comment (cbielow) [inline src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h:53]: typo: `text`

Comment (cbielow) [reply src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h:53]: can you add docs what the text does and where it shows up

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  {
+    x_ = rhs.x_;
+    color_ = rhs.color_;
+  }
+
+  Annotation1DVerticalLineItem::~Annotation1DVerticalLineItem()
+  {
+  }
+
+  void Annotation1DVerticalLineItem::draw(Spectrum1DCanvas * const canvas, QPainter & painter, bool flipped)
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp:64]: coding convention: put `&`, `*` etc next to the type, e.g. `QPainter&`.
Applies to a lot of places here. Should be automatically fixable with clang-format if you happen to have that available.

**Review by cbielow — CHANGES_REQUESTED** [review]
nice. See comments.
Can you provide a screenshot of how it looks like?

**Comment by timosachsenberg** [comment]
image

**Pull request merged**